providers/dynamodb/shedlock-provider-dynamodb2/src/main/java/net/javacrumbs/shedlock/provider/dynamodb2/DynamoDBLockProvider.java

```diff
@@ -78,19 +78,33 @@ public class DynamoDBLockProvider implements LockProvider {
     private final String hostname;
     private final DynamoDbClient dynamoDbClient;
     private final String tableName;
+    private final String partitionKeyName;
 
     /**
      * Uses DynamoDB to coordinate locks
      *
-     * @param dynamoDbClient
-     *            v2 of DynamoDB client
-     * @param tableName
-     *            the lock table name
+     * @param dynamoDbClient v2 of DynamoDB client
+     * @param tableName      the lock table name
      */
     public DynamoDBLockProvider(@NonNull DynamoDbClient dynamoDbClient, @NonNull String tableName) {
         this.dynamoDbClient = requireNonNull(dynamoDbClient, "dynamoDbClient can not be null");
         this.tableName = requireNonNull(tableName, "tableName can not be null");
```

Comment:
I think, initializing the same fields in both constructors is not required. Instead, just use this(dynamoDbClient,tableName, ID) in the old constructor. This would reduce code duplication and would have the same effect.

Reply:
Done, Thank you for suggestion.